src/compile/legend/encode.ts

```diff
@@ -75,9 +74,7 @@ export function symbols(fieldDef: FieldDef, symbolsSpec: any, model: UnitModel,
       {value: cfg.mark.color};
   }
 
-  if (legend.symbolColor !== undefined) {
-    symbols.fill = {value: legend.symbolColor};
-  } else if (symbols.fill === undefined) {
+  if (symbols.fill === undefined) {
```

Comment:
1) Do we even need to set symbols.fill to `cfg.mark.fill`?  I wonder if Vega is doing this for us too, please test.

2) I believe the missing case here is when users set `encoding.color.value` -- the code here should be consistent with `applyColor(AndOpacity)` method